### PR TITLE
feat: add secret and configmap mounting

### PR DESCRIPTION
**Pull request by linuxbandit**
It [used to be impossible](https://github.com/jobtome-labs/ci-templates/blob/v2.16.2/templates/cloudrun.yml#L24) until now:

https://cloud.google.com/sdk/gcloud/reference/run/deploy#--set-secrets

To use it I report the docs:

```
Specify secrets to mount or provide as environment variables. Keys starting with a forward slash '/' are mount paths. All other keys correspond to environment variables. The values associated with each of these should be in the form SECRET_NAME:KEY_IN_SECRET; you may omit the key within the secret to specify a mount of all keys within the secret. For example: '--update-secrets=/my/path=mysecret,ENV=othersecret:key.json' will create a volume with secret 'mysecret' and mount that volume at '/my/path'. Because no secret key was specified, all keys in 'mysecret' will be included. An environment variable named ENV will also be created whose value is the value of 'key.json' in 'othersecret'
```

Therefore the proposed solution is to write something on the pipeline like:

```
# file .gitlab-ci.yml
import: ...
stages: ...
variables:
  ...
  SECRET_MOUNTS: "/secrets=my-app-serviceaccounts"
  ...
```

where we can express both the mount folder (/secrets) and the name of the secret to mount. In this example, as per docs, all keys in the serviceaccount are mounted, so we can have a secret made of 

```
---
apiVersion: v1
kind: Secret
type: Opaque
metadata:
  name: my-app-serviceaccounts
  namespace: my-serverless-ns
data:
  bucket-sa.json: <b64>
  pubsub-sa.json: <b64>
```

and it's up to the app to take the right file

### CODE REVIEW

File: templates/cloudrun.yml [hunk 1]
```diff
@@ -104,6 +95,29 @@
         COMMAND_TO_RUN+=( --set-env-vars=${ENV} )
       fi
 
+      if [ -n "${SECRET_YAML}" ]; then
+
+        echo
+        echo "-> ... also deploying kubernetes secrets"
+        echo
+
+        echo "${SECRET_YAML}" > kube/secret.yaml
```
Comment (gadiener) [inline templates/cloudrun.yml:104]: Base64 decode is missing. Could we move the temporary file out from the repository path? e.g. /tmp/secret

File: templates/cloudrun.yml [hunk 2]
```diff
@@ -104,6 +95,29 @@
...
+        COMMAND_TO_RUN+=( --set-secrets=[${SECRET_MOUNTS}] )
+      fi
+
+      if [ -f "kube/configmap.yaml" ]; then
+
+        echo
+        echo "-> ... also deploying kubernetes configmaps"
+        echo
+
+        kubectl apply -f kube/configmap.yaml --namespace=${NAMESPACE}
```
Comment (gadiener) [inline templates/cloudrun.yml:116]: I think that we must add a variable for the configmap path, kube/configmap.yaml could be the default value

**Comment by gadiener** [comment]
You should add separated variables for quality and production in the root files

File: templates/cloudrun.yml (hunk 2 above)
Comment (linuxbandit) [reply templates/cloudrun.yml]: do you think this is solved this way?

File: templates/cloudrun.yml (hunk 1 above)
Comment (linuxbandit) [reply templates/cloudrun.yml]: done

**Pull request merged**